gossip/gossip/algo/pull.go

```diff
@@ -319,11 +319,18 @@ func (engine *PullEngine) OnRes(items []string, nonce uint64) {
 }
 
 func (engine *PullEngine) newNONCE() uint64 {
-	n := uint64(0)
-	for {
-		n = util.RandomUInt64()
-		if !engine.outgoingNONCES.Exists(n) {
+	engine.lock.Lock()
+	defer engine.lock.Unlock()
+
+	maxAttempts := 100
+	for i := 0; i < maxAttempts; i++ {
+		n := util.RandomUInt64()
+		if n != 0 && !engine.outgoingNONCES.Exists(n) {
 			return n
 		}
 	}
+
```

Comment:
We shouldn't try 100 attempts, but more attempts, like 1000,000. 

and if we still fail, then we should loop until we find a nonce that is not in use using the crypto rand.

Reply:
For the second part it can result in slow nonces if this happens right?